lib/Backend/FlowGraph.cpp

```diff
@@ -172,18 +172,21 @@ FlowGraph::Build(void)
 
     bool assignRegionsBeforeGlobopt = this->func->HasTry() && (this->func->DoOptimizeTry() ||
         (this->func->IsSimpleJit() && this->func->hasBailout) ||
-        this->func->IsLoopBodyInTryFinally());
+        (this->func->IsLoopBodyInTryFinally() && this->func->GetJITFunctionBody()->DoJITLoopBody()));
+
+    bool createNonExceptionFinally = this->func->HasFinally() && (this->func->DoOptimizeTry() ||
+        (this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody()));
```

Comment:
>DoJITLoopBody [](start = 72, length = 13)

I think using this->func->IsLoopBody() would make more sense here. Similarly at two instances below. DoJITLoopBody() also does not guarantee that we are jitting a loop body at this point.

Reply:
I realized I was using this->func->IsSimpleJit() && this->func->GetJITFunctionBody()->DoJITLoopBody(), to mean this->func->IsSimpleJit() && this->func->hasBailout